cmake/scripts/common/ArchSetup.cmake

```
@@ -148,8 +148,24 @@ if(NOT DEFINED NEON OR NEON)
 endif()
 
 if(NOT MSVC)
-  add_options(ALL_LANGUAGES ALL_BUILDS "-Wall" "-Wdouble-promotion" "-Wmissing-field-initializers")
-  add_options(ALL_LANGUAGES DEBUG "-g" "-D_DEBUG")
+  # these options affect all code built by cmake including external projects.
+  add_options(ALL_LANGUAGES ALL_BUILDS
+    -Wall
+    -Wdouble-promotion
+    -Wmissing-field-initializers
+    -Wsign-compare
+  )
+  add_options(ALL_LANGUAGES DEBUG
+    -g
+    -D_DEBUG
+  )
+
+  # these options affect only core code
+  set(CORE_COMPILE_OPTIONS
```

let's allow the user to override:
```suggestion
  if(NOT CORE_COMPILE_OPTIONS)
    set(CORE_COMPILE_OPTIONS
```

should be sufficient, without adding an explicit option

---

I'm fine with that 👍